app/ds/user.go

```diff
@@ -90,6 +93,75 @@ func Delete(userID uint) (err error) {
 	return
 }
 
+func ValidateFacebookToken(logRequest models.UserFacebook) (valid bool,err error) {
+	valid = false
+	resp, err := http.Get("https://graph.facebook.com/v2.5/"+logRequest.FacebookID+"?access_token="+logRequest.FacebookToken);
+	if err != nil {
+        return 
+    }
+    body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return 
+	}
+	var facebookResponse models.FacebookToken
+	err = json.Unmarshal(body, &facebookResponse)
+
+    if facebookResponse.Error !=nil {
+    	valid = false;
+    	return
+    }
+    valid = facebookResponse.FacebookID == logRequest.FacebookID
+    return 
+
+}
+
+func LoginFacebook(logRequest models.UserFacebook) (login map[string]interface{}, err error) {
+	var user models.User
+	var valid bool
+	valid, err =ValidateFacebookToken(logRequest)
+	if valid == false {
+		err = errors.New(`{"facebook":"token-invalid"}`);
+	}
+	if err!=nil {
+		return
+	}
+	err = userSource.Find("email", logRequest.FacebookID).One(&user)
+	if err != nil {
+		err = nil;
+		user.Email =  logRequest.FacebookID;
+		user.FirstName = "";
+		user.LastName = "";
+		user.Answers = logRequest.Answers;
```

Comment:
The answers currently don't get sent to the User API on login, however I think we should do this and also add it to the general login.

However the 'result' values within the answers body need to be separately saved in the total_footprint field. (see [UpdateAnswers()](https://github.com/arbolista-dev/cc-user-api/blob/master/app/controllers/user.go#L171))